src/Icons/src/Registry/CacheIconRegistry.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Icons\Registry;
+
+use Symfony\Component\HttpKernel\CacheWarmer\CacheWarmerInterface;
+use Symfony\Contracts\Cache\CacheInterface;
+use Symfony\UX\Icons\Exception\IconNotFoundException;
+use Symfony\UX\Icons\IconRegistryInterface;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ *
+ * @internal
+ */
+final class CacheIconRegistry implements IconRegistryInterface, CacheWarmerInterface
+{
+    /**
+     * @param IconRegistryInterface[] $registries
+     */
+    public function __construct(private \Traversable $registries, private CacheInterface $cache)
+    {
+    }
+
+    public function get(string $name, bool $refresh = false): array
+    {
+        return $this->cache->get(
+            sprintf('ux-icon-%s', str_replace([':', '/'], ['-', '-'], $name)),
+            function () use ($name) {
+                foreach ($this->registries as $registry) {
+                    try {
+                        return $registry->get($name);
+                    } catch (IconNotFoundException) {
+                        // ignore
+                    }
+                }
+
+                throw new IconNotFoundException(sprintf('The icon "%s" does not exist.', $name));
+            },
+            beta: $refresh ? \INF : null,
+        );
+    }
+
+    public function getIterator(): \Traversable
+    {
+        foreach ($this->registries as $registry) {
+            yield from $registry;
+        }
+    }
+
+    public function count(): int
+    {
+        $count = 0;
+
+        foreach ($this->registries as $registry) {
+            $count += \count($registry);
```

Comment:
What is the goal / usage ?

Reply:
Originally, it was to provide a total for the `ux:icons:warm-cache`'s progress bar: so  "a way to get a count of all currently registered icons"

Reply:
Ok, i see. Love it for the command DX, but as it's not using cached data, it may be a bit "dangerous" to expose i think.

```
if (!empty($registry)) {
```

Would that not call "count" somehow ?

Reply:
> but as it's not using cached data

This is the case for `->getIterator()` as well. As it's not currently used, we can remove `count()` but need to keep `getIterator()`.

Reply:
I think TwigBundle uses specific code for that in commands, but on the other hand i'm not a big fan of having that kind of logic hardcoded in a command, requiring to duplicate it for the "lint" command (that we will need)

Reply:
I've refactored to remove countable as it isn't used.